fdbserver/worker.actor.cpp

```diff
@@ -431,16 +432,23 @@ void updateCpuProfiler(ProfilerRequest req) {
 			break;
 		}
 		break;
+	case ProfilerRequest::Type::GPROF_HEAP:
+#if defined(__linux__) && defined(USE_GPERFTOOLS) && !defined(VALGRIND)
+		HeapProfilerDump("User triggered heap dump");
+#endif
+		break;
 	}
 }
 
 ACTOR Future<Void> runProfiler(ProfilerRequest req) {
 	if (req.action == ProfilerRequest::Action::RUN) {
 		req.action = ProfilerRequest::Action::ENABLE;
 		updateCpuProfiler(req);
```

Comment:
You probably need to rename updateCpuProfiler to something that doesn't have CPU in the name now?

Reply:
updateCpuProfiler() is changed to only be invoked in runCpuProfiler().